APP/app.js

```diff
@@ -1,44 +1,61 @@
 const express = require("express");
-const { getTopics, getEndpoints, getArticlesByID, getArticles, getApi, getCommentsByArticleID, postCommentByArticleID } = require("./controller");
+const {
+  getTopics,
+  getEndpoints,
+  getArticlesByID,
+  getArticles,
+  getApi,
+  getCommentsByArticleID,
+  postCommentByArticleID,
+  patchCommentsByArticleID,
+} = require("./controller");
 const app = express();
 const descriptions = require("../endpoints.json");
 
 app.use(express.json());
 
-app.get("/api/", getApi)
+app.get("/api/", getApi);
 
 app.get("/api/topics", getTopics);
 
-app.get("/api/articles", getArticles)
+app.get("/api/articles", getArticles);
 
 app.get("/api/articles/:article_id", getArticlesByID);
 
-app.get("/api/articles/:article_id/comments", getCommentsByArticleID)
+app.get("/api/articles/:article_id/comments", getCommentsByArticleID);
 
-app.post("/api/articles/:article_id/comments", postCommentByArticleID)
+app.post("/api/articles/:article_id/comments", postCommentByArticleID);
+
+app.patch("/api/articles/:article_id", patchCommentsByArticleID);
 
 app.use((err, req, res, next) => {
-    if (err.code === "22P02") {
-      res.status(400).send({ msg: "Invalid ID type" });
-    } 
-    if(err.code === "23502") {
-      res.status(400).send({msg: "Bad request"})
-    }
-    if(err.code === "23503" && err.constraint === 'comments_author_fkey') {
-      res.status(400).send({msg: "Username does not exist"})
-    }
-    if(err.code === "23503" && err.constraint === 'comments_article_id_fkey') {
-      res.status(400).send({msg: "ID does not exist"})
-    }
-      else next(err);
-  });
+  if (
+    err.code === "22P02" &&
+    err.where === "unnamed portal parameter $1 = '...'"
+  ) {
+    res.status(400).send({ msg: "Bad request, invalid format" });
+  }
+  if (err.code === "22P02") {
```

Comment:
I'm not sure that all of these are strictly necessary, is this just so that you can give back different error messages under the same PSQL error code? If so, it's fine, but I think you gave yourself extra work when giving either not found or bad request for a 400 or 404 would be fine too